storage/copy.go

```diff
@@ -126,7 +126,7 @@ func (c *Copier) Run(ctx context.Context) (attrs *ObjectAttrs, err error) {
 		if err != nil {
 			return nil, err
 		}
-		c.RewriteToken = res.token
+		req.token = res.token
```

Comment:
You actually need to leave the update of `c.RewriteToken` as well -- this is an exported field of `Copier` and the user might use this to resume the copy if it's interrupted.